x-pack/plugins/security_solution/public/common/components/header_actions/actions.tsx

```diff
@@ -111,25 +114,48 @@ const ActionsComponent: React.FC<ActionProps> = ({
     );
   }, [ecsData, eventType]);
 
+  const [visualizationInFlyoutEnabled] = useUiSetting$<boolean>(
+    ENABLE_VISUALIZATIONS_IN_FLYOUT_SETTING
+  );
+
+  const { navigateToAnalyzer } = useNavigateToAnalyzer({
```

Comment:
I am thinking if It makes sense to delegate the responsibility of opening `Analyzer` either in tab or in flyout to `useNavigateToAnalyzer`.

With that logic becomes consolidated. but i am not too strongly opiniated about it. Just something to think about.

Reply:
@logeekal thank you for reviewing! i could see your point, the intention of the hooks was to reduce the duplication of `openFlyout` calls. i would think when it comes to removing the existing analyzer overlay, the code removal would be the same in either places. But I'll think more about it!

Given there is [another PR](https://github.com/elastic/kibana/pull/195693) depending on this one, I will merge it as is

Reply:
makes sense. 👍